cmd/profiles.go

```diff
@@ -0,0 +1,139 @@
+Once a new profile is created, it can be specified with the -p flag, or the ELASTIC_PACKAGE_PROFILE environment variable.
+User profiles are not overwritten on upgrade of elastic-stack, and can be freely modified to allow for different stack configs.`
+
+	profileCommand := &cobra.Command{
+		Use:   "profiles",
+		Short: "Manage stack config profiles",
+		Long:  profilesLongDescription,
+	}
+
+	profileNewCommand := &cobra.Command{
+		Use:   "new",
+		Short: "Create a new profile",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if len(args) == 0 {
+				return errors.New("new requires an argument")
+			}
+			newProfileName := args[0]
+
+			fromName, err := cmd.Flags().GetString(cobraext.ProfileFromFlagName)
+			if err != nil {
+				return cobraext.FlagParsingError(err, cobraext.ProfileFromFlagName)
+			}
+			options := profile.Options{
+				Name:        newProfileName,
+				FromProfile: fromName,
+			}
+			err = profile.CreateProfile(options)
+			if err != nil {
+				return errors.Wrapf(err, "error creating profile %s from profile %s", newProfileName, fromName)
+			}
+
+			fmt.Printf("Created profile %s from %s.\n", newProfileName, fromName)
+
+			return nil
+		},
+	}
+	profileNewCommand.Flags().String(cobraext.ProfileFromFlagName, "default", cobraext.ProfileFromFlagDescription)
+
+	profileDeleteCommand := &cobra.Command{
+		Use:   "delete",
+		Short: "Delete a profile",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("delete requires an argument")
+			}
+			profileName := args[0]
+
+			err := profile.DeleteProfileFromDefaultLocation(profileName)
+			if err != nil {
+				return errors.Wrap(err, "error deleting profile")
+			}
+
+			fmt.Printf("Deleted profile %s\n", profileName)
+
+			return nil
+		},
+	}
+
+	profileListCommand := &cobra.Command{
+		Use:   "list",
+		Short: "List available profiles",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			loc, err := locations.NewLocationManager()
+			if err != nil {
+				return errors.Wrap(err, "error fetching profile")
+			}
+			profileList, err := profile.FetchAllProfiles(loc.StackDir())
+			if err != nil {
+				return errors.Wrap(err, "error listing all profiles")
+			}
+
+			header := []string{"Name", "Date Created", "User", "elastic-package version", "Path"}
+			for _, headerVal := range header {
+				fmt.Printf("%-30s ", headerVal)
+			}
+			fmt.Printf("\n")
+			for _, iter := range profileList {
+				fmt.Printf("%-30s %-30s %-30s %-30s %-30s\n", iter.Name, iter.DateCreated.Format(time.RFC3339), iter.User, iter.Version, iter.From)
+			}
+			return nil
+		},
+	}
+
+	profileCommand.AddCommand(profileNewCommand, profileDeleteCommand, profileListCommand)
+
+	return cobraext.NewCommand(profileCommand, cobraext.ContextGlobal)
+
+}
+
+func lookupEnv() string {
+	env := os.Getenv(cobraext.ProfileNameEnvVar)
```

Comment:
I would be for keeping this env var here (in `cmd`) and leave a string placeholder in `cobraext` (`%s`). Let's not mix these places if not necessary.